.buildkite/pipeline.yml

```diff
@@ -64,6 +64,33 @@ steps:
       - github_commit_status:
           context: "Unit Tests"
 
+  #################
+  # UI Tests
+  #################
+  - group: "🔬 UI Tests"
```

Comment:
- Moved those two UI Tests before the "Lint Translations" step, so that they are next to the "Build for Testing" and that all the related "Run {Unit|UI} Tests" steps stay together
 - Switched to [use a Group step](https://buildkite.com/docs/pipelines/group-step) to group the two related UI Tests together, making the list of steps shown in the Build header/summary look neater. I thought about putting the Unit Test in the same logical group, but figured it would probably be too much grouping, while grouping the 2 steps calling the exact same `.sh` script but for iPhone vs iPad made more sense.

<img width="1157" alt="image" src="https://user-images.githubusercontent.com/216089/176284306-cf4b70b0-a25b-4865-8700-fcb519f59949.png">

Reply:
Just realized there's an opportunity to similarly group the 2 "Installable Build" steps together. Will push a commit doing that as well later.

Reply:
Ended up deciding to separate that change (pipeline reorg) in a dedicated PR, to avoid polluting this one and to keep the two different changes separate.

This part is now implemented in https://github.com/wordpress-mobile/WordPress-iOS/pull/18954, which builds on top of this PR.